src/core/border_router/routing_manager.cpp

```diff
@@ -3681,6 +3681,13 @@ Error RoutingManager::PdPrefixManager::Process(const Ip6::Nd::RouterAdvertMessag
     {
         DiscoveredPrefixTable::Entry entry;
 
+
+        if (aMessage->GetHeader().GetPreferredLifetime()==0)
+        {
+            entry.mPrefix.Clear();
```

Comment:
What is the point of clearing `entry` if we `continue`?

`entry` is scoped to within the `for()` loop body, so on `continue` a new one will be created (objects within `for` scope are constructed/destructed per iteration)

Reply:
Found the handle Goodbye RA is already implemented in below step, close this PR

   if (currentPrefixUpdated && mPrefix.IsDeprecated())
    {
        LogInfo("PdPrefixManager: Prefix %s is deprecated", mPrefix.GetPrefix().ToString().AsCString());
        mPrefix.Clear();
        Get<RoutingManager>().ScheduleRoutingPolicyEvaluation(kImmediately);
    }